dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/ApplicationRegistry.java

```diff
@@ -31,14 +37,30 @@
  *       deployed that contains multiple web modules, an OpenAPI document will only be generated for the first Web
  *       Module that generates an OpenAPI document.
  */
+@Component(configurationPolicy = ConfigurationPolicy.IGNORE, service = ApplicationRegistry.class)
 public class ApplicationRegistry {
 
     private static final TraceComponent tc = Tr.register(ApplicationRegistry.class);
     
-    private static final ApplicationRegistry INSTANCE = new ApplicationRegistry();
+    private static ApplicationRegistry INSTANCE;
```

Comment:
Perhaps we could get rid of this static INSTANCE? Use something like this to look it up instead?
```
        Bundle bundle = FrameworkUtil.getBundle(ApplicationRegistry.class);
        ApplicationRegistry applicationRegistry = AccessController.doPrivileged((PrivilegedAction<ApplicationRegistry>) () -> {
            BundleContext bCtx = bundle.getBundleContext();
            ServiceReference<ApplicationRegistry> svcRef = bCtx.getServiceReference(ApplicationRegistry.class);
            return svcRef == null ? null : bCtx.getService(svcRef);
        });
```

Reply:
We could, and we could also probably get the `BundleContext` in a static initializer, rather than looking it up every time, though we would still need the PrivilegedAction to get the instance and the code is more complicated.

Another option would be to remove the method altogether. The only place that uses the `getInstance` method should now be the servlet which serves the OpenAPI document, so we could have that get hold of the bundle context (which is made available in the servlet context) and look up the service itself.

However, I'm not sure it's worth it to look up a component which is a singleton.